benchmark/src/grpc_helpers.rs

```diff
@@ -92,25 +92,65 @@ fn check_ac_response(resp: &ProtoSubmitTransactionResponse) -> bool {
 /// Ignore but count both gRPC-failed submissions and AC-rejected TXNs.
 pub fn submit_and_wait_txn_requests(
     client: &AdmissionControlClient,
-    txn_requests: Vec<SubmitTransactionRequest>,
+    txn_requests: Vec<LoadRequest>,
     submit_rate: u64,
 ) -> Vec<ProtoSubmitTransactionResponse> {
-    let const_rate_iter = ConstantRate::new(submit_rate, txn_requests.into_iter());
-    let futures: Vec<_> = const_rate_iter
-        .filter_map(|req| {
-            OP_COUNTER.inc("submit_txns"); // This counter doesn't care if submit succeed or not.
-            match client.submit_transaction_async_opt(&req, get_default_grpc_call_option()) {
-                Ok(future) => Some(future),
+    let mut read_futures = vec![];
+    let mut write_futures = vec![];
+    for load_request in ConstantRate::new(submit_rate, txn_requests.into_iter()) {
+        match load_request {
+            LoadRequest::WriteRequest(txn_req) => {
+                match client.submit_transaction_async_opt(&txn_req, get_default_grpc_call_option())
+                {
+                    Ok(future) => write_futures.push(future),
+                    Err(e) => {
+                        OP_COUNTER.inc(&format!("submit_txns.{:?}", e));
+                        error!("Failed to send gRPC request: {:?}", e);
+                    }
+                }
+            }
+            LoadRequest::ReadRequest(ldg_req) => {
+                match client
+                    .update_to_latest_ledger_async_opt(&ldg_req, get_default_grpc_call_option())
+                {
+                    Ok(future) => read_futures.push(future),
+                    Err(e) => {
+                        OP_COUNTER.inc(&format!("submit_txns.{:?}", e));
+                        error!("Failed to send gRPC request: {:?}", e);
+                    }
+                }
+            }
+        }
+        OP_COUNTER.inc("submit_txns");
+    }
+
+    // Process read load's responses. Main thread won't join, hence not blocked by this thread.
+    let read_stream = stream::futures_unordered(read_futures);
```

Comment:
as the function gets longer, I think it's better to extract `read_futures` and `write_futures` processing to dedicated functions.

Reply:
Sure. Extracted to `wait_read/write_futures()`.